### ISSUE TITLE
False positive too-few-public-methods for TypedDict in Python 3.9

### ISSUE THREAD

**Issue by wshanks** [PyCQA/pylint#4180]
This is a follow up to #1927, #2607, and #4072. In Python 3.9, TypedDict was changed from a class to a function with metaclass magic. This change caused worse problems initially (fixed by #4072), but now pylint gives a warning about too few public methods when subclassing TypedDict to create a new type as suggested in the documentation (https://docs.python.org/3/library/typing.html#typing.TypedDict). A work around is to use one of the alternate TypedDict syntax forms in which TypedDict is used as a function instead of a class.

### Steps to reproduce

Given a file `a.py`:
```python
"Demo"
import typing


class Coordinate(typing.TypedDict):
    "This is my type"
    x: float
    y: float
```

### Current behavior

Result of `pylint a.py`:
```
************* Module test
test.py:5:0: R0903: Too few public methods (0/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

### Expected behavior

No warnings.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2                                                                                  
astroid 2.5.1                                                                                 
Python 3.9.2
```

**Comment by cdce8p** [comment]
Yeah, with https://github.com/PyCQA/astroid/pull/915 TypedDicts are now inferred as classes and thus all default class checks do apply. It might indeed make sense to disable `too-few-public-methods` if the class inherits from TypedDict.

**Comment by cdce8p** [comment]
Opened #4191 to fix it
